lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -549,14 +549,17 @@ public static function get_user_data() {
 				isset( $decoded_data['isGlobalStylesUserThemeJSON'] ) &&
 				$decoded_data['isGlobalStylesUserThemeJSON']
 			) {
-				unset( $decoded_data['isGlobalStylesUserThemeJSON'] );
 				$config = $decoded_data;
 			}
 		}
 
 		/** This filter is documented in wp-includes/class-wp-theme-json-resolver.php */
-		$theme_json   = apply_filters( 'wp_theme_json_data_user', new WP_Theme_JSON_Data_Gutenberg( $config, 'custom' ) );
-		$config       = $theme_json->get_data();
+		$theme_json = apply_filters( 'wp_theme_json_data_user', new WP_Theme_JSON_Data_Gutenberg( $config, 'custom' ) );
+		$config     = $theme_json->get_data();
+
+		// Needs to be set for schema migrations of user data.
+		$config['isGlobalStylesUserThemeJSON'] = true;
+
 		static::$user = new WP_Theme_JSON_Gutenberg( $config, 'custom' );
```

Comment:
This property needs to be preset for migrations to run. Migrations are run on both `WP_Theme_JSON_Data_Gutenberg` and `WP_Theme_JSON_Gutenberg` constructors.